### ISSUE TITLE
`./gradlew build` fails with implicit dependency errors

### ISSUE THREAD

**Issue by pramod-knidal** [cortinico/ktfmt-gradle#157]
## 🐛 Describe the bug

`./gradlew build` command fails with the following error

```
* What went wrong:
Some problems were found with the configuration of task 'app:ktfmtFormatMain' (type 'KtfmtFormatTask').
  - Gradle detected a problem with the following location: '<project-dir>/src/main/java'.
    
    Reason: Task 'app:ktfmtCheckMain' uses this output of task 'app:ktfmtFormatMain' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task 'app:ktfmtFormatMain' as an input of 'app:ktfmtCheckMain'.
      2. Declare an explicit dependency on 'app:ktfmtFormatMain' from 'app:ktfmtCheckMain' using Task#dependsOn.
      3. Declare an explicit dependency on 'app:ktfmtFormatMain' from 'app:ktfmtCheckMain' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '<project-dir>/src/main/kotlin'.
    
    Reason: Task 'app:ktfmtCheckMain' uses this output of task 'app:ktfmtFormatMain' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task 'app:ktfmtFormatMain' as an input of 'app:ktfmtCheckMain'.
      2. Declare an explicit dependency on 'app:ktfmtFormatMain' from 'app:ktfmtCheckMain' using Task#dependsOn.
      3. Declare an explicit dependency on 'app:ktfmtFormatMain' from 'app:ktfmtCheckMain' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.

```

## ⚠️ Current behavior

`./gradlew build` fails with error

## ✅ Expected behavior

`./gradlew build` should not throw any warnings/errors related to `ktfmt-gradle` plugin.

## 💣 Steps to reproduce

1. Include ktfmt plugin 0.12.0 in an android library project `id 'com.ncorti.ktfmt.gradle' version '0.12.0'`
2. run `./gradlew build` from terminal

## 📱 Tech info
 - Device: Mac M1
 - OS: Mac 13.4
 - Library Version: 0.12.0
 - Gradle Version: 8.1.1

## Notes

I found this issue when I was trying to generate a release build using 'maven-publish' which threw errors similar to the one above but it was asking me to include the implicit dependency between 'sourceReleaseJar` task and `ktfmtFormatRelease`.

A workaround for now would be appreciated.

**Comment by WhosNickDoglio** [comment]
I _think_ the error message is giving you a workaround? You need to make the relationship between tasks more explicit and could use `dependsOn` or `mustRunAfter` to do that.

Adding the following _should_ work.

```kotlin
tasks.named("ktfmtCheckMain") {
    dependsOn("ktfmtFormatMain")
}
```

**Comment by pramodshri-tgsys** [comment]
> I _think_ the error message is giving you a workaround? You need to make the relationship between tasks more explicit and could use `dependsOn` or `mustRunAfter` to do that.
> 
> Adding the following _should_ work.
> 
> ```kotlin
> tasks.named("ktfmtCheckMain") {
>     dependsOn("ktfmtFormatMain")
> }
> ```

Yeah. Tried several things similar to the one you noted. It was throwing an error saying `ktfmtCheckMain` task not found. In gradle, the ordering matters. So, tried putting that logic in several places until it worked. After it worked, there was another similar error saying `sourcesReleaseJar` should depend on `ktfmtFormatMain`. After I fixed that, there was another similar one. 

I thought having these dependencies explicitly defined in the plugin code would be a good solution. So, posted an issue here.

**Comment by cortinico** [comment]
@pramodshri-tgsys Thanks for the report

> I thought having these dependencies explicitly defined in the plugin code would be a good solution. So, posted an issue here.

The two tasks should not depend on each other. Could you provide a reproducer where you're having this behavior as I can't reproduce it in my testing?

**Comment by pramodshri-tgsys** [comment]
@cortinico I don't have a reproducer. It was happening in a proprietary project when I tried to publish an `aar` for an android library project using `gradle plugin for maven publish`. It happens only when you try to publish a release aar.

You note that the two tasks should not depend on each other. However, adding the code that @WhosNickDoglio has put in his comment will get you past the error... to a new error...

**Comment by simonhauck** [comment]
This issue is a long time not active. If you still think it is important, then please verify the issue still exists with the current ktfmt-gradle version and
- comment/update this issue or 
- open a new issue

If not, I would close this issue in about a week.
